src/backend/distributed/operations/shard_cleaner.c

```diff
@@ -241,19 +243,19 @@ TryDropOrphanedShards(bool waitForLocks)
 	}
 	PG_END_TRY();
 
-	return droppedShardCount;
+	return droppedResourceCount;
 }
 
 
 /*
- * DropOrphanedShardsForCleanup removes resources that were marked for cleanup by operation.
+ * DropOrphanedResourcesForCleanup removes resources that were marked for cleanup by operation.
  * It does so by trying to take an exclusive lock on the resources. If the lock cannot be
  * obtained it skips the resource and continues with others.
  * The resource that has been skipped will be removed at a later iteration when there are no
  * locks held anymore.
  */
 static int
-DropOrphanedShardsForCleanup()
+DropOrphanedResourcesForCleanup()
```

Comment:
Refactoring idea (for a follow up PR): DropOrphanedResourcesForCleanup, FinalizeOperationNeedingCleanupOnSuccess and FinalizeOperationNeedingCleanupOnFailure seem to share a lot of code in their function body. I feel like we could abstract that a bit.

Reply:
Makes sense. We can consider that. Added to my todo list